### PR TITLE
Reorganizing ActiveFedora:File code

### PR DESCRIPTION
**Pull request by awead**
This splits out some of the code in ActiveFedora::File into relevant modules for better organization and to
make the File class overall lighter and more easy to read.

### CODE REVIEW

File: lib/active_fedora/file/fedora_attributes.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+module ActiveFedora::File::FedoraAttributes
```
Comment (jcoyne) [inline lib/active_fedora/file/fedora_attributes.rb]: I like this module. I'm not so keen on the name. I don't think that Fedora has anything do do with whether or not a File has a mime_type, name, digest, etc. Can we just have `AF::File::Attributes`?

File: lib/active_fedora/file/streaming.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+module ActiveFedora::File::Streaming
+  # @param range [String] the Range HTTP header
+  # @return [Stream] an object that responds to each
+  def stream(range = nil)
+    uri = URI.parse(self.uri)
+    FileBody.new(uri, headers(range, authorization_key))
+  end
+
+  # @return [String] current authorization token from Ldp::Client
+  def authorization_key
```
Comment (jcoyne) [inline lib/active_fedora/file/streaming.rb]: I think this should be private.

File: lib/active_fedora/file/fedora_attributes.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    response.map(&:object)
+  end
+
+  protected
+
+    def default_mime_type
+      'text/plain'
+    end
+
+    def fetch_mime_type
```
Comment (jcoyne) [inline lib/active_fedora/file/fedora_attributes.rb]: I think these two methods can be private.  Also, can we change the name of one of them so they both either use `_from_headers` or both don't?

File: lib/active_fedora/file.rb [hunk 4]
```diff
@@ -111,25 +119,10 @@ def remote_content
...
-    def original_name
-      @original_name ||= fetch_original_name_from_headers
-    end
-
-    def digest
-      response = metadata.ldp_source.graph.query(predicate: ActiveFedora::RDF::Fcrepo4.digest)
-      response.map(&:object)
-    end
-
     def persisted_size
```
Comment (jcoyne) [inline lib/active_fedora/file.rb]: I would think this belongs with the original_name, digest, mime_type, etc.

File: lib/active_fedora/file.rb [hunk 5]
```diff
@@ -210,26 +199,12 @@ def readonly?
...
-        URI.decode(m[:filename])
-      end
-
-      def fetch_mime_type
-        ldp_source.head.headers['Content-Type']
-      end
-
     private
 
       def links
```
Comment (jcoyne) [inline lib/active_fedora/file.rb]: `links` should probably go to File::FedoraAttributes too.

**Comment by awead** [comment]
@jcoyne it hath been amended and rebased to thy biddings.

**Pull request merged**